tests/phpunit/tests/user.php

```diff
@@ -605,43 +605,215 @@ public function test_count_many_users_posts() {
 	}
 
 	/**
-	 * Tests salted cache functionality for count_many_users_posts().
+	 * Ensure the second and subsequent calls to count_many_users_posts() are cached.
 	 *
-	 * @ticket 63405
+	 * @ticket 63045
 	 */
-	public function test_count_many_users_posts_salted_cache() {
+	public function test_count_many_users_posts_is_cached() {
+		$user_1 = self::$user_ids[0];
+		$user_2 = self::$user_ids[1];
 
-		$author_id = self::factory()->user->create( array( 'role' => 'author' ) );
-		$editor_id = self::factory()->user->create( array( 'role' => 'editor' ) );
+		// Create posts for both users.
+		self::factory()->post->create( array( 'post_author' => $user_1 ) );
+		self::factory()->post->create( array( 'post_author' => $user_2 ) );
 
-		// Create posts for both users
-		self::factory()->post->create( array( 'post_author' => $author_id ) );
-		self::factory()->post->create( array( 'post_author' => $editor_id ) );
+		// Warm the cache.
+		count_many_users_posts( array( $user_1, $user_2 ), 'post', false );
 
-		// Test cache hit
-		$counts1 = count_many_users_posts( array( $author_id, $editor_id ), 'post', false );
-		$counts2 = count_many_users_posts( array( $author_id, $editor_id ), 'post', false );
-		$this->assertSame( $counts1, $counts2 );
-		$this->assertSame( '1', $counts1[ $author_id ] );
-		$this->assertSame( '1', $counts1[ $editor_id ] );
+		// Ensure cache is hit for second call.
+		$start_queries = get_num_queries();
+		count_many_users_posts( array( $user_1, $user_2 ), 'post', false );
+		$end_queries = get_num_queries();
+		$this->assertSame( $end_queries - $start_queries, 0, 'No database queries expected for second call to count_many_users_posts()' );
```

Comment:
The first arg is the `$expected`:

```suggestion
		$this->assertSame( 0, $end_queries - $start_queries, 'No database queries expected for second call to count_many_users_posts()' );
```